### PR TITLE
Cache client

### PR DESCRIPTION
**Pull request by tharvik**
Depend on https://github.com/c4dt/lightarti-rest-ios/pull/9

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,9 +5,10 @@ all: directory
 
 directory:
 	mkdir directory
-	export URL=$$( curl --silent "https://api.github.com/repos/c4dt/lightarti-directory/releases/latest" | \
-		jq -r '.assets[0].browser_download_url' ); \
-	wget -q -O- $$URL | tar -C directory -xz
+	curl --silent "https://api.github.com/repos/c4dt/lightarti-directory/releases/latest" | \
```
Comment (ineiti) [inline Makefile:8]: Why not download directly from

https://github.com/c4dt/lightarti-directory/releases/latest/download/directory-cache.tgz

?

**Review by ineiti — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (tharvik) [reply Makefile]: I used the method already present in the Makefile but yeah, using the direct URL is way easier

**Review by ineiti — APPROVED** [review]

File: lightarti-rest-ios-test/BackgroundCall.swift [hunk 2]
```diff
@@ -8,32 +8,29 @@
...
-                let resp = try callArti(dict_dir: dict_dir,
-                    method: .GET, url: "https://www.c4dt.org/index.html");
-                if let str = String(data: resp.body, encoding: .utf8){
-                    self.reply = str;
-                } else {
-                    self.reply = "Couldn't decode answer";
-                }
+                let response = try Client
+                    .init(dict_dir: Bundle.main.resourcePath! + "/directory")
+                    .send(request: .init(url: URL.init(string: "https://example.com/")!));
```
[the remark this answers is not in the document]
Comment (tharvik) [reply lightarti-rest-ios-test/BackgroundCall.swift:29]: It doesn't at the moment, the exit timeouts for an unknown reason.

[the remark this answers is not in the document]
Comment (ineiti) [reply lightarti-rest-ios-test/BackgroundCall.swift:29]: Do we have an issue opened for this?

**Pull request merged**